packages/destination-actions/src/destinations/tiktok-audiences/addUser/index.ts

```diff
@@ -43,6 +43,21 @@ const action: ActionDefinition<Settings, Payload> = {
           }
         }
       }
+    },
+    audience_id: async (request, { payload }) => {
+      try {
+        const tiktok = new TikTokAudiences(request)
+
+        return tiktok.fetchAudiences(payload.selected_advertiser_id)
```

Comment:
For the catch block to work, this statement needs to be awaited.
```suggestion
        return await tiktok.fetchAudiences(payload.selected_advertiser_id)
```

Reply:
Another option is chaining catch and returning a value.
```suggestion
        return tiktok.fetchAudiences(payload.selected_advertiser_id).catch(err => { 
           return {
          choices: [],
          error: {
            message: JSON.stringify(err),
            code: '500'
          }
        }
        }
        })
```